### PR TITLE
Added tasks and launch files for vscode, to configure, build and debug

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
         {
             "label": "Configure",
             "type": "shell",
-            "command": "./configure --enable-ccache  --enable-experimental-plugins --enable-example-plugins --enable-test-tools --enable-debug --enable-werror",
+            "command": "autoreconf -if; ./configure --enable-ccache  --enable-experimental-plugins --enable-example-plugins --enable-test-tools --enable-debug --enable-werror",
```
Comment (bneradt) [inline .vscode/tasks.json:21]: A minor thing, but I suggest `&&` to separate the commands in case autoreconf fails for some reason.

Comment (bryancall) [reply .vscode/tasks.json]: 👍 Yeah, I decided to do that and a few more things in the lasted commit.

File: .vscode/tasks.json [hunk 2]
```diff
@@ -3,6 +3,20 @@
...
+        {
+            "label": "Autoreconf",
+            "type": "shell",
+            "command": "autoreconf -if",
+            "problemMatcher": []
+        },
+        {
+            "label": "Configure",
+            "type": "shell",
+            "command": "./configure --prefix=/opt/ats --enable-ccache  --enable-experimental-plugins --enable-example-plugins --enable-test-tools --enable-debug --enable-werror",
```
Comment (calavera) [inline .vscode/tasks.json:15]: unless root access is absolutely necessary, I would not set the prefix in /opt. It also has the inconvenient of being global. I don't know if it's possible to use the workspace variable in this string, but something like `${workspaceFolder}/target` might be better.

This is not a strong opinion, so feel free to not change anything.

Comment (ezelkow1) [reply .vscode/tasks.json:15]: Id be +1 on some env variable for it if possible, we tend to generate everything to use /opt/trafficserver here, so having that be customizable would be good.  Or maybe just have an optional env var insertion that people could throw in stuff like "--prefix=/your/ats/location --with-openssl=/your/ssl/location", so you can get custom things you use into the build

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (randall) [inline .vscode/launch.json:2]: can these comments be removed? They make this file invalid JSON

File: .vscode/tasks.json (hunk 2 above)
Comment (bryancall) [reply .vscode/tasks.json]: @ezelkow1 I am playing around with it to see if I can set an environment variable to override the default

File: .vscode/launch.json (hunk 3 above)
Comment (bryancall) [reply .vscode/launch.json]: 👍

File: .vscode/tasks.json (hunk 2 above)
Comment (bryancall) [reply .vscode/tasks.json]: @ezelkow1 I added the environment variable ATS_VSCODE_CONFIGURE, so you can override configuration options.  It looks like the last option wins in autoconf.

**Comment by zwoop** [comment]
Ship it.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**